### PR TITLE
#768 Add size to setOptions

### PR DESCRIPTION
**Pull request by 3rdvision**
| - | - |
| --- | --- |
| Issue | https://github.com/ripe-tech/ripe-white/issues/768 |
| Decisions | Add size to setOptions as suggested in https://github.com/ripe-tech/ripe-commons-pluginus/pull/222#issuecomment-780702882 |

### CODE REVIEW

File: src/js/base/ripe.js [hunk 1]
```diff
@@ -668,6 +669,7 @@ ripe.Ripe.prototype.setOptions = function(options = {}) {
     this.usePrice = this.options.usePrice === undefined ? !this.noPrice : this.options.usePrice;
     this.noDiag = this.options.noDiag === undefined ? false : this.options.noDiag;
     this.useDiag = this.options.useDiag === undefined ? !this.noDiag : this.options.useDiag;
+    this.size = this.options.size || undefined;
```
Comment (gcandal) [inline src/js/base/ripe.js:672]: Use `null` as fallback, like the others

File: src/js/base/ripe.js [hunk 2]
```diff
@@ -668,6 +669,7 @@ ripe.Ripe.prototype.setOptions = function(options = {}) {
     this.usePrice = this.options.usePrice === undefined ? !this.noPrice : this.options.usePrice;
     this.noDiag = this.options.noDiag === undefined ? false : this.options.noDiag;
     this.useDiag = this.options.useDiag === undefined ? !this.noDiag : this.options.useDiag;
+    this.size = this.options.size || null;
```
Comment (gcandal) [inline src/js/base/ripe.js:672]: Default to `1000`: https://github.com/ripe-tech/ripe-sdk#options

Comment (3rdvision) [reply src/js/base/ripe.js]: I have this defaulting to null in other places
https://github.com/ripe-tech/ripe-sdk/blob/master/src/js/visual/image.js#L53
https://github.com/ripe-tech/ripe-sdk/blob/master/src/js/visual/configurator-prc.js#L51

should I also default to 1000 instead?

Comment (gcandal) [reply src/js/base/ripe.js]: No need I believe, they'll just inherit from here.

**Review by gcandal — APPROVED** [review]

**Comment by joamag** [comment]
This should be part of the configurator options and not the ripe instance... also we already have resolution change support in ripe white

image

**Comment by gcandal** [comment]
@joamag:

image

image

Even though it's under `Configurator` on ripe-white, it's actually calling the RIPE instance method: `this.$bus.bind("size_change", size => self.ripe.setSize(size));`

Why shouldn't this be part of `setOptions` if `size` is an option?

**Comment by joamag** [comment]
> @joamag:
> 
> image
> 
> image
> 
> Even though it's under `Configurator` on ripe-white, it's actually calling the RIPE instance method: `this.$bus.bind("size_change", size => self.ripe.setSize(size));`
> 
> Why shouldn't this be part of `setOptions` if `size` is an option?

Seems to be a bad decision of the past, let's leave things as is and not extend the bad decision of the past.

**Pull request closed without merging**